partcad/src/partcad/ai_feature_file.py

```
@@ -0,0 +1,92 @@
+            self.suffix = ".png"
+        elif lower_url.endswith((".jpg", ".jpeg")):
+            self.is_image = True
+            self.suffix = ".jpg"
+        elif lower_url.endswith(".mp4"):
+            self.is_video = True
+            self.suffix = ".mp4"
+        elif lower_url.endswith(".pdf"):
+            self.is_pdf = True
+            self.suffix = ".pdf"
+
+    def __repr__(self):
```

Ruff (0.8.2)

Missing return type annotation for special method `__repr__`

Add return type annotation: `str`

(ANN204)